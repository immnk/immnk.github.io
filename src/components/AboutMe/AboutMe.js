import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import ScrollAnimation from 'react-animate-on-scroll';

import css from './AboutMe.css';

const AboutMe = ({ className }) => {
  return (
    <section className={classNames(css.root, className)}>
      <ScrollAnimation animateIn="fadeIn" delay={500} initiallyVisible={false}>
        <div className={css.container}>
          <h3 className={css.header}>Hey, Mani here...! Nice to meet you.</h3>
          {/* eslint no-irregular-whitespace: 0 */}
          <p className={css.description}>
            I am a frontend developer having <strong>​7+</strong> years of experience​ in{' '}
            <code>​JavaScript, ReactJS, angular, HTML5, CSS3​</code>. I have solid experience with
            ​NodeJS, ExpressJS, webpack​. I review code on pull requests raised on a shared monorepo
            built with <code>lerna</code>. I also maintain CICD pipelines for code using Jenkins
            &#38; travis. I also established code quality checks with SonarQube. I have considerable
            REST apis developed for local apps. My github account reflects the same. Please reach it
            at{' '}
            <a target="_blank" href="https://github.com/immnk" rel="noreferrer noopener">
              ​https://github.com/immnk
            </a>
          </p>
        </div>
      </ScrollAnimation>
    </section>
  );
};

AboutMe.defaultProps = {
  className: undefined
};

AboutMe.propTypes = {
  className: PropTypes.string
};

export default AboutMe;
