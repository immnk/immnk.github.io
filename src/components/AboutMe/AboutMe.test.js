import React from 'react';
import { shallow } from 'enzyme';

import AboutMe from './AboutMe';

const renderApp = (props) => {
  const defaultProps = {
    className: 'my-classname'
  };

  return <AboutMe {...defaultProps} {...props} />;
};

const shallowRenderApp = (props) => shallow(renderApp(props));

describe('Test App', () => {
  it('render app', () => {
    const actual = shallowRenderApp();
    expect(actual).toMatchSnapshot();
  });
});
