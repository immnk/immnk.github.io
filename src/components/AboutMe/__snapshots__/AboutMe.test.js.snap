// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App render app 1`] = `
<section
  className="root my-classname"
>
  <ScrollAnimation
    animateIn="fadeIn"
    animateOnce={false}
    animatePreScroll={true}
    delay={100}
    duration={1}
    initiallyVisible={false}
    offset={150}
  >
    <div
      className="container"
    >
      <h3
        className="header"
      >
        Hey, Mani here...! Nice to meet you.
      </h3>
      <p
        className="description"
      >
        I am a frontend developer having 
        <strong>
          ​7+
        </strong>
         years of experience​ in
         
        <code>
          ​JavaScript, ReactJS, angular, HTML5, CSS3​
        </code>
        . I have solid experience with ​NodeJS, ExpressJS, webpack​. I review code on pull requests raised on a shared monorepo built with 
        <code>
          lerna
        </code>
        . I also maintain CICD pipelines for code using Jenkins & travis. I also established code quality checks with SonarQube. I have considerable REST apis developed for local apps. My github account reflects the same. Please reach it at
         
        <a
          href="https://github.com/immnk"
          rel="noreferrer noopener"
          target="_blank"
        >
          ​https://github.com/immnk
        </a>
      </p>
    </div>
  </ScrollAnimation>
</section>
`;
